[cxx-interop] [nfc] Remove swift namespace from SwiftShims in C++ mode.

zoecarver opened the pull request:
Most SwiftShims were put in the swift namespace in C++ mode which broke certain things when importing them in a swift file with C++ interop enabled. This was OK when they were only imported as part of the swift runtime but, now they are used in C++ mode both in the swift runtime and when C++ interop is enabled.

gribozavr reviewed (approved):
To be more specific, the `Swift` serialized module file contains references to symbols in SwiftShims headers. Since the standard library is built in default mode (without C++ interop) the symbols in SwiftShims headers are in the global namespace. When we enable C++ interop, it affects SwiftShims headers, too, and the symbols can't be found in the global namespace anymore. Therefore, the compiler can't deserialize certain parts of the `Swift` module.

@zoecarver I'd appreciate if you could put that into your commit description for future readers.

gribozavr commented:
@swift-ci Please test

gottesmm commented:
@gribozavr I would prefer if we just fix this here now. A force push doesn't hurt anyone and commit msgs are important;

zoecarver commented:
@gottesmm I'm in the process of rebasing/fixing the build errors now. Once that's done, I'll squash the commits and update the message.

zoecarver committed:
[cxx-interop] [nfc] Remove swift namespace from SwiftShims in C++ mode.

Most SwiftShims were put in the swift namespace in C++ mode which broke certain things when importing them in a swift file in C++ mode. This was OK when they were only imported as part of the swift runtime but, now they are used in C++ mode both in the swift runtime and when C++ interop is enabled.

This broke when C++ interop was enabled because the `Swift` module contains references to symbols in the SwiftShims headers which are built without C++ interop enabled (no "swift" namespace). But, when C++ interop is enabled, the SwiftShims headers would put everything in the swift namespace meaning the symbols couldn't be found in the global namespace. Then, the compiler would error when trying to deserialize the Swift module.

zoecarver commented:
@gottesmm @gribozavr I've updated the commit message.

Sorry for the delay. I had to rebuild everything after updating checkouts, then had to update Xcode and rebuild everything again 🤦

zoecarver commented:
@swift-ci please test.

zoecarver commented:
@swift-ci please test.

zoecarver commented:
@swift-ci please test windows platform.

zoecarver commented:
@swift-ci please test Windows platform.

zoecarver commented:
@swift-ci please smoke test

zoecarver commented:
@swift-ci please smoke test.

zoecarver commented:
@swift-ci Please smoke test Linux platform

zoecarver commented:
@swift-ci please smoke test Linux platform.

zoecarver commented:
@swift-ci please smoke test Linux platform

zoecarver commented:
@swift-ci please smoke test OS X platform

zoecarver commented:
@swift-ci please smoke test OS X platform

zoecarver commented:
I've updated the commit message. @gottesmm @gribozavr unless you want me to change the commit message further, I'm going to plan to commit this tomorrow.

gribozavr commented:
LGTM, including the commit message! Please merge when you're ready.
